File: app/src/main/java/se/gustavkarlsson/aurora_notifier/android/dagger/modules/ContextModule.java
package se.gustavkarlsson.aurora_notifier.android.dagger.modules;

import android.content.Context;

import dagger.Module;
import dagger.Provides;
import dagger.Reusable;

@Module
public class ContextModule {
	private final Context context;

	public ContextModule(Context context) {
		this.context = context;
	}

	@Provides
	@Reusable
	Context provideContext() {
		return context;
	}
}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/dagger/modules/ContextModule.kt
package se.gustavkarlsson.skylight.android.dagger.modules

import android.content.Context

import dagger.Module
import dagger.Provides
import dagger.Reusable

@Module
class ContextModule(
        private val context: Context
) {

    @Provides
    @Reusable
    fun provideContext(): Context = context
}
